internal/pkg/deploy/deploy.go

```diff
@@ -87,6 +95,82 @@ func NewStore(sessProvider SessionProvider, store ConfigStoreClient) (*Store, er
 	return s, nil
 }
 
+// Pipeline is a deployed pipeline.
+type Pipeline struct {
+	appName      string
+	resourceName string
+	isLegacy     bool
+}
+
+// NewPipeline initiates a new pipeline. Note that this function should only be used for unit tests.
+func NewPipeline(appName, resourceName string, isLegacy bool) Pipeline {
+	return Pipeline{
+		appName:      appName,
+		resourceName: resourceName,
+		isLegacy:     isLegacy,
+	}
+}
+
+// HumanName returns the human-friendly name of the deployed pipeline.
+// For example: bad-goose (legacy)
+func (p Pipeline) HumanName() string {
+	if p.isLegacy {
+		return fmt.Sprintf("%s (legacy)", p.resourceName)
+	}
+	return strings.TrimPrefix(p.resourceName, fmt.Sprintf(pipelineNamePrefix, p.appName))
+}
+
+// Name returns the name of the deployed pipeline.
+func (p Pipeline) Name() string {
+	if p.isLegacy {
+		return p.resourceName
+	}
+	return strings.TrimPrefix(p.resourceName, fmt.Sprintf(pipelineNamePrefix, p.appName))
+}
+
+// ResourceName returns the resource name of the deployed pipeline.
+func (p Pipeline) ResourceName() string {
+	return p.resourceName
+}
+
+// PipelineStore fetches information on deployed pipelines.
+type PipelineStore struct {
+	appName string
+	getter  resourceGetter
+}
+
+// NewPipelineStore returns a new PipelineStore.
+func NewPipelineStore(appName string, getter *rg.ResourceGroups) *PipelineStore {
```

Comment:
can we use an interface as input instead of a specific data type type `*rg.ResourceGroups`?